### PR TITLE
fix(openai_compatible): omit rerank authorization without an API key

### PR DESCRIPTION
**Pull request by 1018053166**
Related to https://github.com/langgenius/dify-official-plugins/issues/1724

## Summary

Omit the `Authorization` header from OpenAI-compatible rerank requests when `api_key` is missing or empty.

The current implementation sends `Bearer None` or an empty Bearer value, which causes gateways for unauthenticated rerank endpoints to reject otherwise valid requests.

## Root cause

`OAICompatRerankModel._invoke` constructs the Bearer header unconditionally.

The same method is used by both model invocation and credential validation, so fixing it there covers both paths.

## Changes

- Build the required `Content-Type` header first.
- Add `Authorization` only when `api_key` is non-empty.
- Add a focused regression test for missing, empty, and non-empty API keys.

## Compatibility

Non-empty API keys produce the same header as before.

Missing, `None`, and empty keys no longer produce malformed authentication headers.

The request URL, body, timeout, response parsing, and score normalization are unchanged.

No dependency, schema, manifest, or public API changes are introduced.

## Scope

This PR fixes the shared SDK `OAICompatRerankModel`.

The current official `openai_api_compatible` plugin overrides this method and requires a separate follow-up.

The related issue should remain open until that override is fixed as well.

## Test plan

- [x] `uv run pytest tests/interfaces/model/openai_compatible/test_rerank.py tests/interfaces/model/test_rerank.py`
- [x] `just check`
- [x] `just test`

### CODE REVIEW

File: python/dify_plugin/interfaces/model/openai_compatible/rerank.py [hunk 1]
```diff
@@ -55,9 +55,13 @@ def _invoke(
 
         url = server_url
         headers = {
-            "Authorization": f"Bearer {credentials.get('api_key')}",
             "Content-Type": "application/json",
         }
+        
+        # 只有当api_key不为空时才添加Authorization头
```
Comment (crazywoola) [inline python/dify_plugin/interfaces/model/openai_compatible/rerank.py]: Please make sure the comment in English.

**Review by Yeuoly — DISMISSED** [review]
LGTM

**Review by WH-2099 — APPROVED** [review]
